sympy/functions/special/delta_functions.py

```diff
@@ -159,6 +159,31 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, arg):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol, oo
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((oo, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((oo, Eq(x - 5, 0)), (0, True))
+
+           >>> y = Symbol('y')
+
+           >>> DiracDelta(y).rewrite(Piecewise)
+           DiracDelta(y)
+
+        """
+        from sympy import oo
```

Comment:
Yeah sure, it can be. But every time I get confused whether to make an import a top-level one or not. Can you please explain me a little bit about this ?